src/reps/array.js

```diff
@@ -19,184 +19,114 @@ const DOM = React.DOM;
  * Renders an array. The array is enclosed by left and right bracket
  * and the max number of rendered items depends on the current mode.
  */
-let ArrayRep = React.createClass({
-  displayName: "ArrayRep",
```

Comment:
Okay I found it.
I think react does very little with our functional component, and that might include propTypes validation.
If instead of directly calling my rep `StringRep(props)` I do `React.createElement(StringRep, props)` I do have the warning messages.
Now I must figure out if doing that will impact the performance of Reps.
If it does, I think we might give Flow a try to replace propTypes maybe ?